4_Functions_refactoring_module_3.py

```diff
@@ -0,0 +1,68 @@
+import re
+
+# Function to normalize header (capitalize first letter)
+def normalize_header(text):
+    return text.split(':\n\t')[0].capitalize()
+
+# Function to normalize the body text (capitalize first letter of each sentence)
+def normalize_body(text):
+    body = text.split(':\n\t')[1]
+    sentences = body.split('\n\t')
+    return '\n\t'.join('. '.join([s.capitalize() for s in sentence.split('. ')]) for sentence in sentences)
+
+# Function to extract the last word from each sentence
+def extract_last_words(normalized_text):
+    sentences = normalized_text.splitlines()
+    return [sentence.split()[-1][:-1] for sentence in sentences if sentence.split()]
+
+# Function to insert a new sentence made of last words into the second paragraph
+def insert_new_sentence(sentences, new_sentence):
+    if len(sentences) >= 2:
+        sentences[2] += ' ' + new_sentence
+    return sentences
+
+# Function to fix "iZ" to "is" where it's a mistake using regex
+def fix_iz(text):
```

Comment:
It will be better to make this function more generic. Now it can be used only for iz cases.